core/commands/version.go

```diff
@@ -83,3 +88,50 @@ var VersionCmd = &cmds.Command{
 	},
 	Type: VersionOutput{},
 }
+
+type Dependency struct {
+	Path       string
+	Version    string
+	ReplacedBy string
+	Sum        string
+}
+
+var depsVersionCommand = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Shows information about dependencies used for build",
+		ShortDescription: `
+Print out all dependencies and their versions.`,
+	},
+	Type: Dependency{},
+
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		info, ok := debug.ReadBuildInfo()
+		if !ok {
+			return errors.New("no embedded dependency information")
+		}
+		toDependency := func(mod *debug.Module) (dep Dependency) {
+			dep.Path = mod.Path
+			dep.Version = mod.Version
+			dep.Sum = mod.Sum
+			if repl := mod.Replace; repl != nil {
+				dep.ReplacedBy = fmt.Sprintf("%s@%s", repl.Path, repl.Version)
+			}
+			return
+		}
+		res.Emit(toDependency(&info.Main))
+		for _, dep := range info.Deps {
+			res.Emit(toDependency(dep))
+		}
+		return nil
+	},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, dep Dependency) error {
```

Comment:
For some reason, if I change `Dependency` to `*Dependency` here I get no output, no error and the encoder is not being run (tested with `panic`).

Reply:
https://github.com/ipfs/go-ipfs-cmds/pull/155